Add new branch dave-dev-pa for feature, fix bug, and replace cropper component"

DaveTang6 opened the pull request.

madarez commented on components/Providers.jsx:
```diff
@@ -0,0 +1,18 @@
+import { AlertProvider } from './AlertProvider';
+
+
+export function Providers({ children}) {
```
Can we use a different term than children, so that it does not get mixed in the context of the children in the app?

madarez commented on components/LoadingProvider.jsx:
This has a parallel in [this PR](https://github.com/murphy-charitable-foundation/penpalapp-nextjs/pull/52/files#diff-1b4a10a06082d04725b9bb2eb4ef0bc73f79ca5136ab054553ae559c2ec8ffb9).

madarez replied:
@su794 should we go with this then? Have you had the chance to review this PR?

madarez reviewed (commented):
Can we stick to the colour green that the designers chose for the continue buttons?

madarez commented:
Also, I'm not sure, why the mouse cursor turns into a text selector and not a hand when skip button is hovered over:

https://github.com/user-attachments/assets/7384c3b8-8db3-4cfa-80b2-6073177fbf0c

DaveTang6 commented:
> Can we stick to the colour green that the designers chose for the continue buttons?

The button is grey by design at first, and turns green only after the user uploads an image. But let me know if there's been a change to the design and I will adjust it

DaveTang6 commented:
> Also, I'm not sure, why the mouse cursor turns into a text selector and not a hand when skip button is hovered over:
> 
>  Screen.Recording.2025-04-29.at.8.02.14.AM.mov

I updated it. The button is disabled before a photo updated. After uploading, it becomes active. And I also added a pointer style to the text "Skip for now".

madarez commented on app/add-profile/page.js:
Can you call this page `onboarding-profile`

DaveTang6 replied:
Sure, I will update it

madarez commented on components/LoadingProvider.jsx:
Instead of the loading page, can you use the loading spinner in the components folder?

DaveTang6 replied:
Sure thing, I’ll use the spinner.

madarez commented on components/ConfirmProvider.jsx:
Can you use popovers in the components folder instead?

madarez commented on components/AlertProvider.jsx:
Despite impressive toast, we can avoid custom alerts like this.

madarez commented on components/general/compress-image.js:
What if it is video or audio?

DaveTang6 replied:
If it's video or audio, I’d suggest handling compression on the server side instead — it’s more efficient and reliable than doing it on the client.

madarez replied:
Is this actually not dealt with in #81 instead?

madarez commented on components/general/AvatarCropper.jsx:
Can you share this functionality with profile page?

DaveTang6 replied:
Sure

madarez commented on components/general/compress-image.js:
```diff
@@ -0,0 +1,74 @@
+export default async function compressImage(input, options = {}) {
+    const { maxWidth, maxHeight, quality = 0.8 } = options;
+  
+    const imageUrl = typeof input === 'string' ? input : URL.createObjectURL(input);
```
Don't you need to import URL?

DaveTang6 replied:
I believe URL is a global object in the browser, so we don’t need to import it.

DaveTang6 commented on components/general/CountrySelect.jsx:
No problem.

DaveTang6 commented on components/general/compress-image.js:
```diff
@@ -0,0 +1,74 @@
+ * Compress an image (supports File, Blob, or URL input)
+ * @param {File|Blob|string} input - The image source
+ * @param {Object} options - Optional settings
+ * @param {number} [options.maxWidth] - Maximum width (maintains aspect ratio)
+ * @param {number} [options.maxHeight] - Maximum height (maintains aspect ratio)
```
Ok

ebert-c commented on components/general/Dropdown.jsx:
```diff
@@ -49,14 +49,25 @@ export default function Dropdown({
         </button>
         {error && <p className="mt-1 text-sm text-red-500">{error}</p>}
-        <Modal
+        {/* <Modal
```
commented out code

madarez commented on app/onboarding-profile/page.js:
```diff
@@ -108,99 +110,123 @@ export default function OnboardingProfile() {
+      setStorageUrl,
+      onSuccess: (url) => {
+        setStep(1);
+        // showAlert("Your avatar has been saved!");
```
Can you remove dead code here?

madarez commented on app/profile/page.js:
```diff
@@ -393,377 +419,376 @@ export default function EditProfile() {
+                {!!avatar && (<div className=" flex items-center justify-between py-3 text-red-500"
+                  onClick={onImageDelete}
+                >
+                  <span>Delete Photo</span>
```
Delete Photo did not take effect for me when I saved and reloaded my letter home I could still see my own profile photo

madarez commented on app/onboarding-profile/page.js:
```diff
@@ -0,0 +1,309 @@
+    <div className="bg-gray-50 min-h-screen flex flex-col">
+      {<div className="p-6 h-full flex flex-col flex-1">
+        <div className="flex justify-between items-center">
+          <BackButton />
```
How do you deal with unsaved changes that may get lost by pressing back?

madarez commented on components/general/AvatarCropper.jsx:
Does this need to exist here while we have an avatar folder?

studentuser13 commented on tsconfig.json:
Please remove this file

studentuser13 commented:
It would be nice if the camera capture feature worked on desktop, e.g. laptops, web cams. I only looked into this briefly, it seems like this would require a lot of work... is this out of the scope of this PR?

studentuser13 commented:
<img width="1888" height="1111" alt="Screenshot 2026-01-30 at 6 12 50 PM" src="https://github.com/user-attachments/assets/41c48571-a3c5-41ea-a7d7-872d9f558426" />

### I like the ability to resize the selection area that the original component in `main` provides. Giving it a circular share is fairly trivial with CSS. 

### Is it possible to take the Take Photo Option and workflow to a fresh PR using the cropper in `main`?

madarez commented on app/edit-profile-user-image/page.js:
```diff
@@ -86,70 +56,83 @@ export default function EditProfileUserImage() {
+    }
+
+    if (!croppedImage) {
+      alert("请先裁剪图片");
```
??
